File: apps/pass-at-k-study/benchmark/tasks/iris/l4_fault_tolerant_aggregation.cpp
// Task: l4_fault_tolerant_aggregation — matches L4FaultTolerantAggregation.
#include <cmath>
#include <iostream>

static float aggregate(int N) {
  float s = 0.f;
  for (int i = 1; i <= 4; ++i) {
    float v = (i == 3) ? NAN : float(i) * float(N);
    if (std::isfinite(v)) s += v;
  }
  return s;
}

int main() {
  float a = aggregate(10);
  float b = aggregate(100);
  if (std::fabs(a - 70.f) > 1e-3f || std::fabs(b - 700.f) > 1e-3f) {
    std::cout << "FAIL\n";
    return 1;
  }
  std::cout << "PASS\n";
  return 0;
}

File: apps/pass-at-k-study/benchmark/tasks/legate/l4_fault_tolerant_aggregation.py
# Task: l4_fault_tolerant_aggregation — matches L4FaultTolerantAggregation.
import math
import sys


def aggregate(n: int) -> float:
    s = 0.0
    for i in range(1, 5):
        v = float("nan") if i == 3 else float(i * n)
        if math.isfinite(v):
            s += v
    return s


def main() -> int:
    a = aggregate(10)
    b = aggregate(100)
    if abs(a - 70.0) > 1e-3 or abs(b - 700.0) > 1e-3:
        print("FAIL")
        return 1
    print("PASS")
    return 0


if __name__ == "__main__":
    sys.exit(main())
